utils/advanced_rpc_compressor.hh

```diff
@@ -81,7 +81,7 @@ struct compression_algorithm {
 class compression_algorithm_set {
     uint8_t _bitset;
     static_assert(std::numeric_limits<decltype(_bitset)>::digits > compression_algorithm::count());
-    constexpr compression_algorithm_set(uint8_t v) noexcept : _bitset(v) {}
+    constexpr compression_algorithm_set(int v) noexcept : _bitset(static_cast<uint8_t>(v)) {}
```

Comment:
Why accept signed int if it gets truncated to uint8_t?

Reply:
Better cast at the call site, where it's possible to understand why the cast is correct.